packages/devextreme/js/core/utils/ajax_utils.js

```diff
@@ -0,0 +1,200 @@
+    }
+
+    return options.contentType ||
+        defaultContentType;
+};
+
+const getAcceptHeader = function(options) {
+    const dataType = options.dataType || '*';
+    const scriptAccept = 'text/javascript, application/javascript, application/ecmascript, application/x-ecmascript';
+    const accepts = {
+        '*': '*/*',
+        text: 'text/plain',
+        html: 'text/html',
+        xml: 'application/xml, text/xml',
+        json: 'application/json, text/javascript',
+        jsonp: scriptAccept,
+        script: scriptAccept
+    };
+
+    extendFromObject(accepts, options.accepts, true);
+
+    return accepts[dataType] ?
+        accepts[dataType] + (dataType !== '*' ? ', */*; q=0.01' : '') :
+        accepts['*'];
+};
+
+const getRequestHeaders = function(options) {
+    const headers = options.headers || {};
+
+    headers['Content-Type'] = headers['Content-Type'] || getContentTypeHeader(options);
+    headers['Accept'] = headers['Accept'] || getAcceptHeader(options);
+
+    if(!options.crossDomain && !headers['X-Requested-With']) {
+        headers['X-Requested-With'] = 'XMLHttpRequest';
+    }
+    return headers;
+};
+
+const getJsonpOptions = function(options) {
+    if(options.dataType === 'jsonp') {
+        const random = Math.random().toString().replace(/\D/g, '');
+        const callbackName = options.jsonpCallback || 'dxCallback' + Date.now() + '_' + random;
+        const callbackParameter = options.jsonp || 'callback';
+
+        options.data = options.data || {};
+        options.data[callbackParameter] = callbackName;
+
+        return callbackName;
+    }
+};
+
+const getRequestOptions = function(options, headers) {
+    let params = options.data;
+    const paramsAlreadyString = typeof params === 'string';
+    let url = options.url || window.location.href;
+
+    if(!paramsAlreadyString && !options.cache) {
+        params = params || {};
+        params['_'] = Date.now();
+    }
+
+    if(params && !options.upload) {
+        if(!paramsAlreadyString) {
+            params = paramsConvert(params);
+        }
+
+        if(getMethod(options) === 'GET') {
+            if(params !== '') {
+                url += (url.indexOf('?') > -1 ? '&' : '?') + params;
+            }
+            params = null;
+        } else if(headers['Content-Type'] && headers['Content-Type'].indexOf('application/x-www-form-urlencoded') > -1) {
+            params = params.replace(/%20/g, '+');
+        }
+    }
+
+    return {
+        url: url,
+        parameters: params
+    };
+};
+
+const isCrossDomain = function(url) {
+    if(!hasWindow()) {
+        return true;
+    }
+
+    let crossDomain = false;
+    const originAnchor = domAdapter.createElement('a');
+    const urlAnchor = domAdapter.createElement('a');
+
+    originAnchor.href = window.location.href;
+
+    try {
+        urlAnchor.href = url;
+
+        // NOTE: IE11
+        // eslint-disable-next-line no-self-assign
+        urlAnchor.href = urlAnchor.href;
```

Comment:
Could you explain in couple of words why this line is needed?

Reply:
i don't touch this code it's legacy